src/main/java/com/google/devtools/build/docgen/templates/attributes/common/target_compatible_with.html

```diff
@@ -0,0 +1,75 @@
+<p>
+<code>List of <a href="../build-ref.html#labels">labels</a>; optional;</code>
+</p>
+
+<p>
+A list of
+<code><a href="platform.html#constraint_value">constraint_values</a></code>
+that must be present in the target platform for this target. This is in
+addition to any constraints already set by the rule type. Constraints are used
+to restrict the list of available target platforms. If the target platform does
+not satisfy all listed constraints then the target is considered "incompatible"
+and will be skipped for building and testing.
+</p>
+
+<p>
+Targets that transitively depend on incompatible targets are themselves also
+considered incompatible. These will also be skipped for building and testing.
+</p>
+
+<p>
+An empty list signifies that the target is compatible with all platforms.
+<p>
+
+<p>
+The simplest usage of this attribute restricts a target to a single platform.
+The target will not be built for any platform that doesn't satisfy all of the
+constraints.
+<pre class="code">
+cc_library(
+    name = "foo_lib",
+    srcs = "foo_lib.cc",
+    target_compatible_with = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:windows",
+    ],
+)
+</pre>
+</p>
+
+<p>
+Use <code><a href="functions.html#select">select()</a></code> to express more
+complicated restrictions. For example, use an identity to implement basic OR
+logic. The following marks a library compatible with amd64 and arm64, but no
+other platforms.
+<pre class="code">
+cc_library(
+    name = "bar_lib",
+    srcs = "bar_lib.cc",
+    target_compatible_with = select({
+        "@platforms//cpu:x86_64": ["@platforms//cpu:x86_64"],
+        "@platforms//cpu:arm64": ["@platforms//cpu:arm64"],
+        "//conditions:default": ["//:not_compatible"],
```

Comment:
It might be nice having a sentence explaining why this works. Not everyone reading will be super-familiar with selects and may not immediately parse what "magic" is going on here to cause OR semantics.

Reply:
Good idea. Done. Let me know if you like the new section. I also made this example not an identity anymore. An empty list is more readable and has the same functionality.

This now lives in https://docs.bazel.build/versions/master/platforms.html.